docs/environments/gke.md

```diff
@@ -0,0 +1,63 @@
+# Google Kubernetes Engine (GKE) Deployment Guide
+
+Federation v2 can be deployed to and manage [GKE](https://cloud.google.com/kubernetes-engine/) clusters. Since
+Federation v2 requires Kubernetes v1.11 or greater, you must create
+[Alpha Clusters](https://cloud.google.com/kubernetes-engine/docs/concepts/alpha-clusters) and manually specify
+the `cluster-version`. Alpha clusters will no longer be needed once GKE supports Kubernetes `v1.11` or greater. You can
+use the `$ gcloud container get-server-config` command to view the GKE cluster versions available to you.
+The following example deploys two GKE alpha clusters named `cluster1` and `cluster2` using Kubernetes version
+`1.11.2-gke.9`.
+
+```bash
+$ export ZONE=$(gcloud config get-value compute/zone)
+$ gcloud container clusters create cluster1 --zone $ZONE --enable-kubernetes-alpha --cluster-version 1.11.2-gke.9 \
+--no-enable-autorepair --no-enable-autoupgrade
+$ gcloud container clusters create cluster2 --zone $ZONE --enable-kubernetes-alpha --cluster-version 1.11.2-gke.9 \
+--no-enable-autorepair --no-enable-autoupgrade
+```
+**NOTE:** GKE alpha clusters expire after 30 days.
+
+If you are following along with the Federation v2 [User Guide](../userguide.md), change the cluster context names:
+```bash
+$ export CLUSTER1=$(kubectl config get-contexts | grep cluster1 | awk '{print $2}')
```

Comment:
With `print $2` we are grabbing the name of the cluster instead of the context name, even though GCP uses the same for both. This also would break if there are any other context names containing `cluster1` in them that grep finds first. In any case, any reason not to use the original suggestion below?

```bash
export GCP_PROJECT=$(gcloud config list --format='value(core.project)')
kubectl config rename-context gke_${GCP_PROJECT}_${ZONE}_cluster1 cluster1
kubectl config rename-context gke_${GCP_PROJECT}_${ZONE}_cluster2 cluster2
```

Reply:
Thanks for catching this. I'll update and also implement your suggestions: https://github.com/kubernetes-sigs/federation-v2/pull/316#pullrequestreview-166185035